activerecord/lib/active_record/scoping/named.rb

```diff
@@ -151,14 +151,14 @@ def scope(name, body, &block)
               scope = all.scoping { instance_exec(*args, &body) }
               scope = scope.extending(extension) if extension
 
-              scope || all
+              (scope.object_id != self.object_id && scope) || all
```

Comment:
updated!

though we have to do it like `!scope.equal?(self) && scope` so that `scope` gets returned